File: Yandex/Practicum/Algorightms_and_Data_Structures/1_sprint/G.Работа_из_дома/solution.cpp
#include <iostream>
#include <vector>

void solution(int num) {
    /*
    Time Complexity O(N), where N is length of place values
    Memory Complexity O(N), where N is length of place values
    */
    if (!num) {
        std::cout << "0\n";
        return;
    }
    int base = 2;
    std::vector<int> digits;
    while (num) {
        digits.push_back(num % base);
        num /= base;
    }
    int n = digits.size();
    for (int index = n - 1; index >= 0; --index) {
        std::cout << digits[index];
    }
    std::cout << "\n";
};


int main() {
    int num;
    std::cin >> num;
    solution(num);
}

File: Yandex/Practicum/Algorightms_and_Data_Structures/1_sprint/G.Работа_из_дома/solution.py
import sys


def solution(num: int) -> None:
    """
    Time Complexity O(N), where N is length of place values
    Memory Complexity O(N), where N is length of place values
    """
    if not num:
        sys.stdout.write('0\n')
        return
    base: int = 2
    digits: list[int] = []
    while num:
        digits.append(num % base)
        num //= base
    digits_length: int = len(digits)
    for idx in range(digits_length - 1, -1, -1):
        sys.stdout.write(str(digits[idx]))
    sys.stdout.write('\n')


if __name__ == '__main__':
    num: int = int(sys.stdin.readline().rstrip())
    solution(num)
